backend/src/controller/chatController.ts

```diff
@@ -289,6 +325,15 @@ async function handleChatToGPT(req: OpenAiChatRequest, res: Response) {
 		updatedChatHistory = addErrorToChatHistory(updatedChatHistory, errorMsg);
```

Comment:
Both here and on line 320, we are setting updatedChatHistory to add the error message but then doing nothing with that updated value - because we immediately return from this function. This feels like a regression, because before @heatherlogan-scottlogic's PR #743, the original chatHistory was being mutated as a side-effect.

Reply:
Ah yes agreed. Would likely be best to address this in #743 ? As that's where the change was made.

Reply:
changed these instances to 
`req.session.levelState[currentLevel].chatHistory = addErrorToChatHistory(
			updatedChatHistory,
			errorMessage
		);`